### PR TITLE
[FIX] web: buttons with special attrs close dialogs

### PR DESCRIPTION
**Pull request by Iucapad**
This commit fixes the behavior of those customized buttons, used in form view footers, aiming at saving/discarding the changes. The expected behavior is that the dialog closes itself when those buttons are pressed, which was no longer applied since the rewrite of views in Owl.

Also, the newly created value was not selected after saving the record. Now, it selects it immediately if a button with special="save" has been pressed.

A test will be added to verify that dialogs behaves correctly and close themselves when clicking such buttons.

task-3254368

### CODE REVIEW

File: addons/web/static/src/views/view_button/view_button_hook.js [hunk 1]
```diff
@@ -64,7 +64,10 @@ export function useViewButtons(model, ref, options = {}) {
                     enableButtons(getEl(), manuallyDisabledButtons, enableAction);
                     return;
                 }
-                const closeDialog = clickParams.close && env.closeDialog;
+                if (clickParams.special === "save") {
+                    await options.saveButtonClicked(clickParams);
```
Comment (aab-odoo) [inline addons/web/static/src/views/view_button/view_button_hook.js:68]: So before this commit it didn't save either? Or did it save thanks to the "save on leave" feature? If so, doesn't it also save if you click on a special="cancel" button?

File: addons/web/static/tests/views/form/form_view_tests.js [hunk 2]
```diff
@@ -4303,6 +4305,60 @@ QUnit.module("Views", (hooks) => {
...
+            resModel: "partner",
+            resId: 2,
+            arch: `
+                <form>
+                    <sheet><group>
+                    <field name="product_id"/>
+                    </group></sheet>
+                </form>`,
+            mockRPC(route, args) {
+                if (args.method === "get_formview_id") {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form/form_view_tests.js:4335]: I'd assert step here, to record the write rpcs we do

Comment (Iucapad) [reply addons/web/static/tests/views/form/form_view_tests.js]: I improved the test and asserted the "create" and "write" calls.

- First write ABCDE and save it -> create
- Save the form view -> write
- Write XYZ but cancel creation in the dialog -> not created
- form view has nothing to save

File: addons/web/static/src/views/view_button/view_button_hook.js (hunk 1 above)
Comment (Iucapad) [reply addons/web/static/src/views/view_button/view_button_hook.js:68]: The record was created by the dialog, but not applied in the many2one field from the form view (it remained empty).
Now, when created, it is selected right after the dialog closed

**Comment by Iucapad** [comment]
Hello @aab-odoo Can you check this PR? I've made some changes to the test since the last review. I added some assertions

File: addons/web/static/src/views/view_button/view_button_hook.js (hunk 1 above)
Comment (aab-odoo) [reply addons/web/static/src/views/view_button/view_button_hook.js:68]: I still don't understand why you call this function here. I would have assumed that the fix would only be to set closeDialog to true if clickParams.special (what you do line 70)

Comment (aab-odoo) [reply addons/web/static/src/views/view_button/view_button_hook.js:68]: Maybe we should talk about this PR. I'm wondering if the problem isn't in `beforeExecuteActionButton` of form_controller instead (where we save, but where we totally ignore the props saveRecord)

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**